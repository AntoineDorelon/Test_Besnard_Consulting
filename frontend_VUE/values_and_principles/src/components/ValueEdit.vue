<template>
 <div v-if="value">
   <p>{{value.title.charAt(0).toUpperCase() + value.title.slice(1)}}</p>
 </div>
</template>

<script>
export default {
  name: "ValueEdit",
  props: ['value']
}
</script>

<style scoped>

</style>