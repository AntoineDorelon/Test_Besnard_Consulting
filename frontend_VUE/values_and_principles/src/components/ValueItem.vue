<template>
  <div>
    <p>
      {{ value.title.charAt(0).toUpperCase() + value.title.slice(1) }}
    </p>

    <font-awesome-icon class="edit" icon="edit" @click="$emit('value-edit',value.id)" />

    <font-awesome-icon class="trash" icon="trash" @click="$emit('value-delete', value.id)" />

  </div>
</template>

<script>
export default {
  name: "ValueItem",
  props: ['value']
}
</script>

<style scoped>
p {
  text-align: left;
  width: 70%;
  display: inline-block;
  margin-bottom: 0px;
  padding: 5px 0;
}
.edit{
  color: blue;
  cursor: pointer;
  float: end;
  margin-right: 15px;
  font-size: 200%;
}
.trash {
  color: red;
  cursor: pointer;
  float: end;
  margin-left: 15px;
  font-size: 200%;
}
</style>