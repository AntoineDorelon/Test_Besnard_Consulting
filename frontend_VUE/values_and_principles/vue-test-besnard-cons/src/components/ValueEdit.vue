<template>
  <div v-if="value">
    <p>
      <span style="text-decoration: underline">Value :</span> &nbsp;
      <label for="title">{{
        value.title.charAt(0).toUpperCase() + value.title.slice(1)
      }}</label>
      <br />
      <input id="title" placeholder="Title" v-model="localValue.title" />
      <button @click="saveValue()" type="button" class="btn btn-primary">
        Save the modifications
      </button>
    </p>
  </div>
</template>

<script>
export default {
  name: "ValueEdit",
  props: ["value"],
  data() {
    return {
      localValue: { ...this.value },
    };
  },
  watch: {
    value: function (newval, oldval) {
      if (newval !== oldval) {
        this.localValue = { ...this.value };
      }
    },
  },
  methods: {
    saveValue() {
      if (this.value.id) {
        console.log("Value to edit is :", this.value.title);
        fetch(`http://127.0.0.1:8000/api/values/${this.value.id}/`, {
          method: "PUT",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify({ title: this.localValue.title }),
        })
          .then((response) => response.json())
          .then((response) => {
            this.$emit("ValueUpdated");
            console.log(response);
          })
          .catch((error) => console.log(error));
      } else {
        fetch(`http://127.0.0.1:8000/api/values/`, {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify({ title: this.localValue.title }),
        })
          .then((response) => response.json())
          .then((response) => {
            this.$emit("ValueUpdated");
            console.log(response);
          })
          .then((this.value = { title: "" }))
          .catch((error) => console.log(error));
      }
    },
  },
};
</script>

<style scoped>
p {
  text-align: left;
}

button {
  margin-left: 2rem;
  margin-bottom: 1rem;
  margin-top: 1rem;
}
</style>
