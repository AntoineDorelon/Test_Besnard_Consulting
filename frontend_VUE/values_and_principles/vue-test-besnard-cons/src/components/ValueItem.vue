<template>
  <div>
    <p>
      {{ value.title.charAt(0).toUpperCase() + value.title.slice(1) }}
    </p>

    <font-awesome-icon
      class="edit"
      :icon="EditIcon"
      @click="$emit('ValueEdit', value.id)"
    />

    <font-awesome-icon
      class="trash"
      :icon="TrashIcon"
      @click="$emit('ValueDelete', value.id)"
    />
  </div>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { faEdit } from "@fortawesome/free-solid-svg-icons";
import { faTrash } from "@fortawesome/free-solid-svg-icons";

export default {
  name: "ValueItem",
  props: ["value"],
  components: {
    FontAwesomeIcon,
  },
  data() {
    return {
      EditIcon: faEdit,
      TrashIcon: faTrash,
    };
  },
};
</script>

<style scoped>
p {
  text-align: left;
  width: 70%;
  display: inline-block;
  margin-bottom: 0;
  padding: 5px 0;
}

.edit {
  color: blue;
  cursor: pointer;
  float: end;
  margin-right: 15px;
  font-size: 200%;
}

.trash {
  color: red;
  cursor: pointer;
  float: end;
  margin-left: 15px;
  font-size: 200%;
}
</style>
