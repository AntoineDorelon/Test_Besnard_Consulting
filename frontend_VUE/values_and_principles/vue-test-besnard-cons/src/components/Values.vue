<template>
  <div class="list-group">
    <h2>List of Values</h2>
    <small class="text-muted"
      >Number of values : {{ Object.keys(values).length }} / 4</small
    >

    <ValueItem
      class="list-group-item"
      v-for="value in values"
      :key="value.id"
      :value="value"
      @ValueEdit="valueEdit($event)"
      @ValueDelete="valueDelete($event)"
    />
    <br />
    <ValueEdit
      v-if="editedValue"
      :value="editedValue"
      @ValueUpdated="refreshValues()"
    />

    <h5 v-if="Object.keys(values).length < 4">
      <font-awesome-icon class="plus" :icon="PlusIcon" @click="newValue()" />
      Click the "+" icon to add another value
    </h5>
  </div>
</template>

<script>
import ValueItem from "./ValueItem";
import ValueEdit from "./ValueEdit";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { faPlus } from "@fortawesome/free-solid-svg-icons";

export default {
  name: "Values",
  components: {
    ValueEdit,
    ValueItem,
    FontAwesomeIcon,
  },
  data() {
    return {
      values: [],
      editedValue: null,
      PlusIcon: faPlus,
    };
  },
  methods: {
    valueEdit(value_id) {
      console.log("value to edit has id ", value_id);
      this.editedValue = this.values.find((value) => value.id === value_id);
    },
    newValue() {
      this.editedValue = { title: "" };
    },
    valueDelete(value_id) {
      console.log("value to delete has id ", value_id);
      fetch(`http://127.0.0.1:8000/api/values/${value_id}/`, {
        method: "DELETE",
        headers: {
          "Content-Type": "application/json",
        },
      })
        .then(() => {
          this.values = this.values.filter((value) => value.id !== value_id);
        })
        .catch((error) => console.log(error));
    },
    getValues() {
      return fetch("http://127.0.0.1:8000/api/values/", {
        method: "GET",
      }).then((response) => response.json());
    },
    refreshValues: async function () {
      this.values = await this.getValues();
    },
  },
  async created() {
    await this.refreshValues();
  },
};
</script>

<style scoped>
.list-group {
  padding-left: 2rem;
  padding-right: 2rem;
}

h5 {
  text-align: left;
  margin-bottom: 0px;
  /*vertical-align: middle;*/
}

.plus {
  color: green;
  font-size: 200%;
  vertical-align: middle;
  cursor: pointer;
}
</style>
