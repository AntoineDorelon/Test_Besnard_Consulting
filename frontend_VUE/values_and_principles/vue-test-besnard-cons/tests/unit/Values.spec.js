import ValueItem from "../../src/components/ValueItem";
import Values from "../../src/components/Values";
import { createLocalVue, mount } from "@vue/test-utils";
import ValueEdit from "../../src/components/ValueEdit";

describe("Value component with 3 values", () => {
  const mockedValues = [
    { id: 1, title: "first Value" },
    { id: 2, title: "second Value" },
    { id: 3, title: "third Value" },
  ];
  const getValues = jest.fn().mockResolvedValue(mockedValues);

  const wrapper = mount(Values, {
    methods: { getValues },
  });

  test("contains 3 ValueItem components when we passed it through the props", () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.html()).toContain(
      "First Value",
      "Second Value",
      "Third Value"
    );
    //another way to test it is by using the find methods in the child components because toContain is discouraged
    expect(wrapper.findAllComponents(ValueItem).length).toBe(3);
  });

  test("displays the add icon when we have less than 4 values", () => {
    expect(wrapper.html()).toContain('Click the "+" icon to add another value');
  });

  test("displays the edit component when clicking the plus icon", async () => {
    const plusButton = wrapper.find(".plus");
    await plusButton.trigger("click");

    const valueEditComponent = wrapper.findComponent(ValueEdit);
    expect(wrapper.vm.$data.editedValue).toEqual({ title: "" });
    expect(valueEditComponent.exists()).toBe(true);
  });

  async function getAndDisplayEditComponent() {
    wrapper.setData({ editedValue: { title: "" } });
    await wrapper.vm.$nextTick();
    const valueEditComponent = wrapper.findComponent(ValueEdit);
    return valueEditComponent;
  }

  test("passes the value to the local value when the user type in the input", async () => {
    const valueEditComponent = await getAndDisplayEditComponent();
    const textInput = valueEditComponent.find("input");

    textInput.element.value = "valueToAdd";
    await textInput.trigger("input");
    await valueEditComponent.vm.$nextTick();

    expect(valueEditComponent.vm.$data.localValue.title).toBe("valueToAdd");
  });

  test("calls the saveValue method when the user click on the save button", async () => {
    const valueEditComponent = await getAndDisplayEditComponent();
    const spy = jest.spyOn(valueEditComponent.vm, "saveValue");
    const saveButton = valueEditComponent.find("button");

    await saveButton.trigger("click");
    await valueEditComponent.vm.$nextTick();

    expect(spy).toHaveBeenCalled();
  });
});

describe("Value component with 4 values", () => {
  const localVue = createLocalVue();
  const mockedValues = [
    { id: 1, title: "first Value" },
    { id: 2, title: "second Value" },
    { id: 3, title: "third Value" },
    { id: 4, title: "forth Value" },
  ];
  const getValues = jest.fn().mockResolvedValue(mockedValues);

  const wrapper = mount(Values, {
    methods: { getValues },
    localVue: localVue,
  });

  test("doesn't display the add icon or the edit component when we have 4 values", () => {
    expect(wrapper.html()).not.toContain(
      'Click the "+" icon to add another value'
    );
    const ValueEditComponent = wrapper.findComponent(ValueEdit);
    expect(ValueEditComponent.exists()).toBe(false);
  });

  test("displays the Value Edit component when we click the edit icon and placeholder has the value already written", async () => {
    const ValueItems = wrapper.findAllComponents(ValueItem);
    const spy = jest.spyOn(wrapper.vm, "valueEdit");
    await ValueItems.at(1).vm.$emit("ValueEdit", 2);
    expect(spy).toHaveBeenCalled();
    await wrapper.vm.$nextTick();
    const ValueEditComponent = wrapper.findComponent(ValueEdit);
    expect(ValueEditComponent.exists()).toBe(true);
    const textInput = ValueEditComponent.find("input").element.value;
    expect(textInput).toBe("second Value");
  });
});
