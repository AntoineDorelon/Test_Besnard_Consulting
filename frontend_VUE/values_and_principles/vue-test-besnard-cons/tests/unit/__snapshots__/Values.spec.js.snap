// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value component contains 3 ValueItem components when we passed it through the props 1`] = `
<div class="list-group">
  <h2>List of Values</h2> <small class="text-muted">Number of values : 3 / 4</small>
  <div class="list-group-item">
    <p>
      First Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Second Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Third Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div> <br>
  <!---->
  <h5><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="plus svg-inline--fa fa-plus fa-w-14">
      <path fill="currentColor" d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z" class=""></path>
    </svg>
    Click the "+" icon to add another value
  </h5>
</div>
`;

exports[`Value component with 3 values contains 3 ValueItem components when we passed it through the props 1`] = `
<div class="list-group">
  <h2>List of Values</h2> <small class="text-muted">Number of values : 3 / 4</small>
  <div class="list-group-item">
    <p>
      First Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Second Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Third Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div> <br>
  <!---->
  <h5><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="plus" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="plus svg-inline--fa fa-plus fa-w-14">
      <path fill="currentColor" d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z" class=""></path>
    </svg>
    Click the "+" icon to add another value
  </h5>
</div>
`;

exports[`Value component with 4 values displays the Value Edit component when we click the edit icon 1`] = `
<div class="list-group">
  <h2>List of Values</h2> <small class="text-muted">Number of values : 4 / 4</small>
  <div class="list-group-item">
    <p>
      First Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Second Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Third Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div>
  <div class="list-group-item">
    <p>
      Forth Value
    </p> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="edit" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" class="edit svg-inline--fa fa-edit fa-w-18">
      <path fill="currentColor" d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z" class=""></path>
    </svg> <svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="trash" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="trash svg-inline--fa fa-trash fa-w-14">
      <path fill="currentColor" d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z" class=""></path>
    </svg>
  </div> <br>
  <!---->
  <!---->
</div>
`;
